src/types/any.rs

```diff
@@ -146,36 +146,36 @@ impl PyAny {
     where
         O: ToPyObject,
     {
+        /// Mostly the same as ffi::PyObject_RichCompareBool, but this function does not try a == b
         unsafe fn do_compare(
             py: Python,
             a: *mut ffi::PyObject,
             b: *mut ffi::PyObject,
-        ) -> PyResult<Ordering> {
-            let result = ffi::PyObject_RichCompareBool(a, b, ffi::Py_EQ);
-            if result == 1 {
-                return Ok(Ordering::Equal);
-            } else if result < 0 {
+            op: c_int,
+        ) -> PyResult<bool> {
+            let result = ffi::PyObject_RichCompare(a, b, op);
+            if result.is_null() {
                 return Err(PyErr::fetch(py));
```

Comment:
is it okay if you don't call `ffi::PyDECREF(result)` here?

Reply:
Oh, good catch :+1: , thank you.
I actually forgot to call it.